### PR TITLE
fix multiple places where we were returning http.StatusNotFound and should not have been

### PR DESCRIPTION
**Pull request by jim-minter**
One of these is 404->500.  The others are 404->400s.

I've audited the remaining places where we return 404 and believe they are correct.

### CODE REVIEW

File: pkg/frontend/middleware/validate.go [hunk 1]
```diff
@@ -48,50 +48,50 @@ func Validate(env env.Interface, apis map[string]*api.Version) func(http.Handler
...
+					api.WriteError(w, http.StatusBadRequest, api.CloudErrorCodeInvalidLocation, "", "The provided location '%s' is malformed or invalid.", vars["location"])
 					return
 				}
 			}
 
 			if _, found := vars["operationId"]; found {
 				_, err := uuid.FromString(vars["operationId"])
 				if err != nil {
-					api.WriteError(w, http.StatusNotFound, api.CloudErrorCodeInvalidOperationID, "", "The provided operation identifier '%s' is malformed or invalid.", vars["operationId"])
+					api.WriteError(w, http.StatusBadRequest, api.CloudErrorCodeInvalidOperationID, "", "The provided operation identifier '%s' is malformed or invalid.", vars["operationId"])
```
Comment (JackQuincy) [inline pkg/frontend/middleware/validate.go:94]: this should be a 404 if they are asking for the operation

File: pkg/frontend/middleware/validate.go [hunk 2]
```diff
@@ -48,50 +48,50 @@ func Validate(env env.Interface, apis map[string]*api.Version) func(http.Handler
...
-					api.WriteError(w, http.StatusNotFound, api.CloudErrorCodeInvalidResourceType, "", "The resource type '%s' could not be found in the namespace '%s' for api version '%s'.", vars["resourceType"], vars["resourceProviderNamespace"], vars["api-version"])
+					api.WriteError(w, http.StatusBadRequest, api.CloudErrorCodeInvalidResourceType, "", "The resource type '%s' could not be found in the namespace '%s' for api version '%s'.", vars["resourceType"], vars["resourceProviderNamespace"], vars["api-version"])
 					return
 				}
 			}
 
 			if _, found := vars["resourceName"]; found {
 				if !rxResourceGroupName.MatchString(vars["resourceName"]) {
-					api.WriteError(w, http.StatusNotFound, api.CloudErrorCodeResourceNotFound, "", "The Resource '%s/%s/%s' under resource group '%s' was not found.", vars["resourceProviderNamespace"], vars["resourceType"], vars["resourceName"], vars["resourceGroupName"])
+					api.WriteError(w, http.StatusBadRequest, api.CloudErrorCodeResourceNotFound, "", "The Resource '%s/%s/%s' under resource group '%s' was not found.", vars["resourceProviderNamespace"], vars["resourceType"], vars["resourceName"], vars["resourceGroupName"])
```
Comment (JackQuincy) [inline pkg/frontend/middleware/validate.go:79]: if asking for the resource this is a 404.

File: pkg/frontend/middleware/validate.go [hunk 3]
```diff
@@ -48,50 +48,50 @@ func Validate(env env.Interface, apis map[string]*api.Version) func(http.Handler
 			if _, found := vars["subscriptionId"]; found {
 				_, err := uuid.FromString(vars["subscriptionId"])
 				if err != nil {
-					api.WriteError(w, http.StatusNotFound, api.CloudErrorCodeInvalidSubscriptionID, "", "The provided subscription identifier '%s' is malformed or invalid.", vars["subscriptionId"])
+					api.WriteError(w, http.StatusBadRequest, api.CloudErrorCodeInvalidSubscriptionID, "", "The provided subscription identifier '%s' is malformed or invalid.", vars["subscriptionId"])
```
Comment (JackQuincy) [inline pkg/frontend/middleware/validate.go:51]: if this is a call for get the subscription it should be a 404.  I'm less sure on other calls @amanohar  thoughts?

File: pkg/frontend/middleware/validate.go (hunk 1 above)
Comment (jim-minter) [reply pkg/frontend/middleware/validate.go:94]: this is if the operationId passed is not a UUID

File: pkg/frontend/middleware/validate.go (hunk 2 above)
Comment (jim-minter) [reply pkg/frontend/middleware/validate.go:79]: this is validation of the resourceName

File: pkg/frontend/middleware/validate.go (hunk 3 above)
Comment (jim-minter) [reply pkg/frontend/middleware/validate.go:51]: ditto - validation the subscriptionId is a uuid

**Pull request merged**